scripts/benchmark-alert.py

```diff
@@ -0,0 +1,147 @@
+    GITHUB_APP_PRIVATE_KEY - the private key file contents of the GitHub App
+    CONBENCH_URL - the URL to the Conbench server where benchmark results are stored
+"""
+
+
+def parse_args() -> Tuple[str, str, str, float]:
+    parser = argparse.ArgumentParser(
+        description=DESCRIPTION, formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument(
+        "--contender-sha",
+        type=str,
+        required=True,
+        help="The SHA hash of the contender commit. The script will download benchmark "
+        "information about this commit and post a GitHub Check of the full report to "
+        "this commit.",
+    )
+    parser.add_argument(
+        "--merge-commit-message",
+        type=str,
+        required=True,
+        help="If the condender commit is a merge-commit, its message. If not, an empty "
+        "string.",
+    )
+    parser.add_argument(
+        "--z-score-threshold",
+        type=float,
+        required=True,
+        help="The (positive) z-score threshold. Benchmarks with a z-score more extreme "
+        "than this threshold will be marked as regressions.",
+    )
+    args = parser.parse_args()
+
+    repo = os.environ["GITHUB_REPOSITORY"]
+
+    return (
+        repo,
+        args.contender_sha,
+        args.merge_commit_message,
+        args.z_score_threshold,
+    )
+
+
+def _merge_commit_pr_number(commit_message: str) -> Optional[int]:
+    """If this is a merge-commit run, get the number of the PR that was merged by
+    grepping the commit message.
+
+    Other alternatives to this strategy:
+
+    - Hit /repos/{owner}/{repo}/commits/{commit_sha}/pulls to get the number. This
+      doesn't work because of the way facebook-github-bot merges PRs.
+    - Use the GitHub GraphQL API to search through recent "PR closed" events until you
+      find the one that's associated with this commit. This does work but actually feels
+      *more* likely to break than grepping.
+    """
+    if not commit_message:
+        print("Merge-commit message not given; this must be a PR commit")
+        return None
+
+    # Look for e.g. (#123) at the end of the first line
+    res = re.search(r"\(#(\d+)\)$", commit_message.split("\n")[0])
+
+    if not res:
+        print("Could not find the PR number in the following merge-commit message:")
+        print(commit_message)
+        return None
+
+    print(f"Found a PR number (in the given merge-commit message): {repr(res[1])}")
+    return int(res[1])
+
+
+def main(
+    repo: str,
+    contender_sha: str,
+    merge_commit_message: str,
+    z_score_threshold: float,
+):
+    log.setLevel("DEBUG")
+    pipeline_steps = [
+        steps.GetConbenchZComparisonStep(
+            commit_hash=contender_sha, z_score_threshold=z_score_threshold
+        ),
+        steps.GitHubCheckStep(repo=repo),
+    ]
+
+    # Only post a comment if this arg is non-empty (i.e. on merge-commits)
```

Comment:
I chose not to post a comment each time PR benchmarks are run, because PR authors will already see the result of the analysis as a GitHub Check. I think that would get too noisy.

Relatedly, I tested this out and after this PR is merged and it runs for the first time, you'll be able to add a branch protection rule, so you can only merge a PR if there were no performance regressions.

Reply:
I agree about the noise an alternative would be to add one comment and edit it with updated results.

Reply:
sidenote: if the event is `pull_request` the workflow downloads an artifact with the PR number in it which get's set as an output.

Reply:
>an alternative would be to add one comment and edit it with updated results

Good idea, created https://github.com/conbench/conbench/issues/985. I won't do that in this PR.

>if the event is pull_request the workflow downloads an artifact with the PR number in it which get's set as an output

Yep I saw that -- that would be easy to pass as an arg to this script if we wanted to post comments on pull-request runs. Of course it's not populated for merge-commit runs so we have to grep the commit message to figure out the pull request for those. (see my code comments about this)